src/parser/node.js

```diff
@@ -14,8 +14,13 @@ class Node implements NodeBase {
     this.type = "";
     this.start = pos;
     this.end = 0;
-    this.loc = new SourceLocation(loc);
```

Comment:
I would expect a function returning an object to be as fast as just an object (since it will inline it quickly by the engine). Could we try with a function `MakeSourceLocation` which returns the location object.

Otherwise ok for me :+1: